velox/functions/prestosql/aggregates/benchmarks/AverageBenchmark.cpp

```diff
@@ -0,0 +1,177 @@
+    auto nulls = facebook::velox::allocateNulls(size, pool());
+    auto rawNulls = nulls->asMutable<uint64_t>();
+    for (auto i = 0; i < size; i += 4) {
+      bits::setNull(rawNulls, i, true);
+    }
+
+    auto c1 = vectorMaker_.flatVector<int64_t>(
+        size,
+        [](auto row) { return row % 5; },
+        [](auto row) { return row % 23 == 0; });
+    auto c2 = vectorMaker_.flatVector<double>(
+        size,
+        [](auto row) { return row % 5 + 0.1; },
+        [](auto row) { return row % 37 == 0; });
+    return vectorMaker_.rowVector(
+        {vectorMaker_.flatVector<int32_t>(
+             size, [](auto row) { return row % 7; }),
+         BaseVector::wrapInDictionary(nulls, indices, size, c1),
+         BaseVector::wrapInDictionary(nulls, indices, size, c2)});
+  }
+
+  RowVectorPtr makeData(VectorEncoding::Simple encoding) {
+    switch (encoding) {
+      case VectorEncoding::Simple::FLAT:
+        return makeFlatData();
+      case VectorEncoding::Simple::CONSTANT:
+        return makeConstantData();
+      case VectorEncoding::Simple::DICTIONARY:
+        return makeDictionaryData();
+      default:
+        VELOX_UNREACHABLE();
+    }
+  }
+
+  void doRun(
+      const std::vector<std::string>& groupingKeys,
+      const std::vector<std::string>& aggregates,
+      VectorEncoding::Simple encoding) {
+    folly::BenchmarkSuspender suspender;
+
+    const int32_t kLoopSize = 10'000;
+    std::vector<RowVectorPtr> values;
+    for (auto i = 0; i < kLoopSize; ++i) {
+      values.push_back(makeData(encoding));
+    }
+    auto plan = exec::test::PlanBuilder(pool())
+                    .values(values)
+                    .partialAggregation(groupingKeys, aggregates)
+                    .intermediateAggregation()
+                    .finalAggregation()
+                    .planNode();
+
+    std::shared_ptr<folly::Executor> executor{
+        std::make_shared<folly::CPUThreadPoolExecutor>(
+            std::thread::hardware_concurrency())};
+    auto task = exec::Task::create(
+        "avg aggregate benchmark task",
+        core::PlanFragment{plan, core::ExecutionStrategy::kUngrouped, 1, {}},
+        0,
+        std::make_shared<core::QueryCtx>(executor.get()));
+
+    suspender.dismiss();
+
+    int cnt = 0;
+    auto result = task->next();
+    while (result != nullptr) {
+      cnt += result->size();
+      result = task->next();
+    }
+    folly::doNotOptimizeAway(cnt);
+  }
+
+  void runWithFlatInput() {
+    doRun({"c0"}, {"avg(c1)", "avg(c2)"}, VectorEncoding::Simple::FLAT);
+  }
+
+  void runWithConstantInput() {
+    doRun({"c0"}, {"avg(c1)", "avg(c2)"}, VectorEncoding::Simple::CONSTANT);
+  }
+
+  void runWithDictionaryInput() {
+    doRun({"c0"}, {"avg(c1)", "avg(c2)"}, VectorEncoding::Simple::DICTIONARY);
+  }
+};
+
+BENCHMARK(flat) {
+  AverageAggregateBenchmark benchmark;
+  benchmark.runWithFlatInput();
+}
+
+BENCHMARK_RELATIVE(constant) {
+  AverageAggregateBenchmark benchmark;
```

Comment:
AverageAggregateBenchmark benchmark;
should be excluded or alternatively global variable